src/sdk.spec.js

```diff
@@ -1,86 +1,43 @@
+import mswServer from 'api-mocks/msw-server';
+import {mockFormioTranslations, mockLanguageInfoGet} from 'components/LanguageSelection/mocks';
 
-// TODO: look into msw for API mocking instead
-// see https://testing-library.com/docs/react-testing-library/example-intro
-const apiModule = require('api');
-jest.mock('api');
+import {OpenForm} from './sdk.js';
 
 // scrollIntoView is not not supported in Jest
 let scrollIntoViewMock = jest.fn();
 window.HTMLElement.prototype.scrollIntoView = scrollIntoViewMock;
 
-const FORM = {
-  uuid: '81a22589-abce-4147-a2a3-62e9a56685aa',
-  url: 'http://localhost:8000/api/v2/forms/81a22589-abce-4147-a2a3-62e9a56685aa',
-  name: 'MOCKED',
-  active: true,
-  loginRequired: false,
-  loginOptions: [],
-  product: '',
-  slug: 'jest',
-  maintenanceMode: false,
-  showProgressIndicator: true,
-  submissionAllowed: 'yes',
-  literals: {
-    beginText: {resolved: 'begin', value: ''},
-    changeText: {resolved: 'change', value: ''},
-    confirmText: {resolved: 'confirm', value: ''},
-    previousText: {resolved: 'previous', value: ''},
-  },
-  steps: [],
-  explanationTemplate: '',
-  requiredFieldsWithAsterisk: true,
-  autoLoginAuthenticationBackend: '',
-  translationEnabled: true,
-};
-
-const LANGUAGE_INFO = {
-  languages: [
-    {code: 'nl', name: 'Nederlands'},
-    {code: 'en', name: 'English'},
-  ],
-  current: 'nl',
-};
-
-const mockAPICalls = url => {
-  switch (url) {
-    case 'http://localhost:8000/api/v2/forms/81a22589-abce-4147-a2a3-62e9a56685aa': {
-      return FORM;
-    }
-    case 'http://localhost:8000/api/v2/i18n/formio/nl': {
-      return {};
-    }
-    case 'http://localhost:8000/api/v2/i18n/info': {
-      return LANGUAGE_INFO;
-    }
-    default:
-      throw new Error(`Unknown url: ${url}`);
-  }
-};
+const LANGUAGES = [
+  {code: 'nl', name: 'Nederlands'},
+  {code: 'en', name: 'English'},
+];
+
+const apiMocks = [
+  mockFormGet(getForm({translationEnabled: true})),
+  mockLanguageInfoGet(LANGUAGES),
+  mockFormioTranslations,
+];
 
 describe('OpenForm', () => {
   it('should accept a DOM node as languageSelectorTarget', async () => {
+    mswServer.use(...apiMocks);
     const formRoot = document.createElement('div');
     const target = document.createElement('div');
-
-    apiModule.get.mockImplementation(mockAPICalls);
-
     const form = new OpenForm(formRoot, {
-      baseUrl: 'http://localhost:8000/api/v2/',
+      baseUrl: BASE_URL,
       basePath: '',
       formId: '81a22589-abce-4147-a2a3-62e9a56685aa',
       languageSelectorTarget: target,
       lang: 'nl',
     });
 
-    try {
-      await act(async () => await form.init());
-    } catch (e) {
-      throw e; // should not error
-    }
+    await act(async () => await form.init());
 
+    // wait for the loader to be removed when all network requests have completed
+    await waitForElementToBeRemoved(() => within(formRoot).getByRole('status'));
     expect(target).not.toBeEmptyDOMElement();
```

Comment:
the new way of mocking actually exposed that our tests weren't handling the async behaviour properly - since we now have actual IO and non-instance promise resolution, the tests were trying to assert while the loader was still visible.

This is resolved by given the loader an accessible role and querying by that role to wait until the loader is gone and the actual form is visible.

Probably this could also be "wait for form title to be visible", I'm not married to this approach :thinking:

Reply:
This now waits until all `status` role children are gone from the form element, right? I have no idea whether other status elements are imaginable, nor the desire to deep dive into aria right now.